lib/TPP/Dialect/Xsmm/XsmmUtils.cpp

```diff
@@ -148,6 +148,30 @@ FailureOr<BinaryInfo> getBinaryInfo(Value lhs, BinaryFlags lhsFlag, Value rhs,
   return binaryInfo;
 }
 
+FailureOr<int64_t> getLeadingDim(Type type, size_t pos) {
+  // Not shaped type, the leading dimension is the single scalar.
+  if (!isa<ShapedType>(type))
```

Comment:
You don't need `!isa<ShapedType>(type)` since you already have `!isa<MemRefType>(type)`.